### ISSUE TITLE
check-param-names: does not check arrow function properties in interfaces

### ISSUE THREAD

**Issue by LangLangBart** [gajus/eslint-plugin-jsdoc#1657]
## Expected behavior

`check-param-names` should flag mismatched `@param` names on arrow function properties in TypeScript interfaces, the same way it does for method signatures (added in #1251).

## Actual behavior

Only method signature syntax (`TSMethodSignature`) is checked. Arrow function property syntax (`TSPropertySignature`) is silently skipped.

## ESLint Config

```json
// package.json
{
  "dependencies": {
    "eslint": "^9",
    "eslint-plugin-jsdoc": "^62.5.4",
    "typescript": "^5",
    "typescript-eslint": "^8.55.0"
  },
  "scripts": {
    "lint:check": "eslint"
  },
  "type": "module"
}
```

```js
// eslint.config.js
import parser from '@typescript-eslint/parser'
import jsdoc from 'eslint-plugin-jsdoc'

export default [
  {
    files: ['**/*.ts'],
    languageOptions: { parser },
    plugins: {
      jsdoc
    },
    rules: {
      'jsdoc/check-param-names': 'warn'
    }
  }
]
```

## ESLint sample

```js
// index.d.ts
interface Foo {
  /** @param wrongName ... */
  method(name: string): void         // ✅ TSMethodSignature is checked
  /** @param wrongName ... */
  arrow: (name: string) => void      // ❌ TSPropertySignature is not checked
}
```

(Only one warning instead of the expected two.)

```bash
bun run lint:check
~/testing-jsdoc/index.d.ts
  2:1  warning  Expected @param names to be "name". Got "wrongName"  jsdoc/check-param-names

✖ 1 problem (0 errors, 1 warning)
```

## Environment

- Node version: v25.6.1
- ESLint version: v9.39.2
- `eslint-plugin-jsdoc` version: 62.5.4
